### ISSUE TITLE
PoissonNLLLoss for regression

### ISSUE THREAD

**Issue by salvomcl** [tracel-ai/burn#2751]
### Feature description

[`PoissonNLLLoss`](https://pytorch.org/docs/stable/generated/torch.nn.PoissonNLLLoss.html)

### Feature motivation

Hi all, I’ve noticed that this loss function is not available, despite being popular across several industries. 
If you think it’s worth adding, I’d be happy to contribute by implementing it.

Thank you!

**Comment by laggui** [comment]
I'm not super familiar with this loss function, but contributions are always welcomed!

**Comment by salvomcl** [comment]
Great, I'll work on this!
